### PR TITLE
[MERGE][IMP] survey: apply small tweaks on survey views (certification)

### PR DESCRIPTION
**Pull request by awa-odoo**
Purpose
======

This merge implies several small tweaks on the survey views, mainly to add support for the new certification fields.

Show the quizz_passed field in the answers views
-------------------------------------------------------------------

The quizz_passed field is now more important because it determines whether
a certification has been successfully taken or not.

The quizz_passed field is added in:
- The search view of user_input
- The list view of user_input
- The form view of user_input (along with the score)

Fix survey statistics computation
-------------------------------------------

Fix the shared dict in the survey statistics computation that
causes surveys to incorrectly share the same stats when computed in batch.

Add a stat button on survey to show the certified count
------------------------------------------------------------------------

Add a stat button on the survey form view showing the amount
of answers with the quizz_passed field set to True.
It allows to quickly see people that have been granted the certification.

On click, the stat button redirects to the answers list view with the "Quizz passed"
filter defaulted to True.

Task-1936539

### CODE REVIEW

File: addons/survey/views/survey_user_views.xml [hunk 1]
```diff
@@ -13,9 +13,10 @@
                 <filter name="completed" string="Completed" domain="[('state', '=', 'done')]"/>
                 <filter string="Partially Completed" name="partially_completed" domain="[('state', '=', 'skip')]"/>
                 <filter string="New" name="new" domain="[('state', '=', 'new')]"/>
+                <separator groups="base.group_no_one"/>
+                <filter string="Invite" name="invite" domain="[('input_type', '=', 'link')]" groups="base.group_no_one"/>
```
Comment (tde-banana-odoo) [inline addons/survey/views/survey_user_views.xml]: Well I don't like filters in debug mode. Either we keep them either we remove them if they have no use. It is used to see people you invited on a survey form view, different from people that subscribed. Maybe it is not interesting, in which case it should be cleaned in master.

File: addons/survey/models/survey_user.py [hunk 2]
```diff
@@ -32,6 +32,7 @@ class SurveyUserInput(models.Model):
 
     # description
     survey_id = fields.Many2one('survey.survey', string='Survey', required=True, readonly=True, ondelete='cascade')
+    scoring_type = fields.Selection(string="Scoring", related="survey_id.scoring_type")
```
Comment (tde-banana-odoo) [inline addons/survey/models/survey_user.py]: Mmmh ... ok for this one.

File: addons/survey/models/survey_user.py [hunk 3]
```diff
@@ -58,7 +59,7 @@ class SurveyUserInput(models.Model):
     question_ids = fields.Many2many('survey.question', string='Predefined Questions', readonly=True)
     deadline = fields.Datetime('Deadline', help="Datetime until customer can open the survey and submit answers")
 
-    quizz_score = fields.Float("Score for the quiz (%)", compute="_compute_quizz_score", default=0.0)
+    quizz_score = fields.Float("Score (%)", compute="_compute_quizz_score", default=0.0)
```
Comment (tde-banana-odoo) [inline addons/survey/models/survey_user.py]: While we are at it let us remove the default, computed fields should not have default :) .

File: addons/survey/models/survey_survey.py [hunk 4]
```diff
@@ -119,6 +122,22 @@ def _compute_survey_statistic(self):
...
+            read_group_res = self.env['survey.user_input'].read_group(
+                [('survey_id', 'in', certificate_surveys.ids), ('test_entry', '!=', True), ('quizz_passed', '=', True)],
+                [],
+                ['survey_id']
+            )
+            for item in read_group_res:
+                stat[item['survey_id'][0]] += item['survey_id_count']
+
+        for survey in self:
+            survey.certified_count = stat[survey.id] if survey.id in stat else 0
```
Comment (tde-banana-odoo) [inline addons/survey/models/survey_survey.py]: stat.get(survey.id, 0)

File: addons/survey/models/survey_survey.py [hunk 5]
```diff
@@ -105,10 +106,12 @@ def _compute_users_can_signup(self):
     @api.depends('user_input_ids.state', 'user_input_ids.test_entry')
     def _compute_survey_statistic(self):
         stat = dict((cid, dict(invite_count=0, answer_count=0, answer_done_count=0)) for cid in self.ids)
-        UserInput = self.env['survey.user_input']
-        base_domain = ['&', ('survey_id', 'in', self.ids), ('test_entry', '!=', True)]
-
-        read_group_res = UserInput.read_group(base_domain, ['input_type', 'state'], ['survey_id', 'input_type', 'state'], lazy=False)
+        read_group_res = self.env['survey.user_input'].read_group(
```
Comment (tde-banana-odoo) [inline addons/survey/models/survey_survey.py]: Unnecessary diff

File: addons/survey/views/survey_user_views.xml (hunk 1 above)
Comment (awa-odoo) [reply addons/survey/views/survey_user_views.xml]: I removed the "group_no_one" on it.
We will maybe question its use for the next saas but let's keep it for now.

File: addons/survey/models/survey_user.py (hunk 2 above)
Comment (awa-odoo) [reply addons/survey/models/survey_user.py]: :angel:

File: addons/survey/models/survey_user.py (hunk 3 above)
Comment (awa-odoo) [reply addons/survey/models/survey_user.py]: It's cleaned.

File: addons/survey/models/survey_survey.py (hunk 4 above)
Comment (awa-odoo) [reply addons/survey/models/survey_survey.py]: Better indeed.

File: addons/survey/models/survey_survey.py (hunk 5 above)
Comment (awa-odoo) [reply addons/survey/models/survey_survey.py]: I did some cleaning but since this targets stable it should contain minimal diff indeed, I reverted it.

**Comment by awa-odoo** [comment]
@tde-banana-odoo I removed the last commit and saved it in another branch for master. We will discuss the code when the time comes.

All other requested changes were applied.

Thanks for the nice review!

**Comment by awa-odoo** [comment]
@jem-odoo Can you take over on this one?

Thanks :)

File: addons/survey/views/survey_templates.xml [hunk 6]
```diff
@@ -178,7 +178,7 @@
                             </p>
                         </div>
                         <a role="button" class="btn btn-primary btn-lg" t-att-href="'/survey/fill/%s/%s' % (survey.access_token, answer.token)">
-                            Start Survey
+                            Start <t t-esc="'Certification' if survey.certificate else 'Survey'" />
```
Comment (jem-odoo) [inline addons/survey/views/survey_templates.xml]: Use a t-if instead. Otherwise the 2 terms will not be translated (I think).

Comment (awa-odoo) [reply addons/survey/views/survey_templates.xml]: Easier to read anyway. Done!

**Comment by jem-odoo** [comment]
LGTMamadou
@robodoo rebase-merge r+
